### PR TITLE
Create domain viewer tracker

### PR DESCRIPTION
**Pull request by ryonext**
## Background

This is a gem for tracking users in a same domain and multi services.
- Generate cookie with uuid.(It becomes viewer id)
- Associate a user_id in a service and a viewer id in the domain.
### Install

```
gem domain_viewer_tracker"
bundle
rails g domain_viewer_tracker install
rake db:migrate
```
### Setting

open 

`config/initializes/domain_viewer_tracker.rb`

and you can edit,
- cookie_domain
- cookie_key_name
### execute

`include DomainViewerTracker` in your controller, and call `set_viewer_id` for all user access and call `store_viewer_id(your_user_id)` when user log in to your service.

### CODE REVIEW

File: spec/domain_viewer_tracker_spec.rb [hunk 1]
```diff
@@ -1,11 +1,125 @@
...
+
+    context "Viewer record already exist" do
+      let(:user_id) { 1 }
+
+      before do
+        test_cookie[:viewer_id] = "hoge"
+        Viewer.create!(uuid: "hoge", user_id: 1)
+      end
+
+      it "Creates viewer record." do
```
Comment (nozayasu) [inline spec/domain_viewer_tracker_spec.rb]: I think the description is not correct.
I think "Does not create viewer record" is correct.
What do you think?

File: spec/domain_viewer_tracker_spec.rb [hunk 2]
```diff
@@ -1,11 +1,125 @@
...
+    TestModel.new
   end
+  let(:test_cookie) { ActionDispatch::Cookies::CookieJar.new(nil) }
 
-  it 'does something useful' do
-    expect(false).to eq(true)
+  describe "#set_viewer_id" do
+    subject { testModel.send(:set_viewer_id) }
+
+    let(:testModel) do
```
Comment (nozayasu) [inline spec/domain_viewer_tracker_spec.rb]: Looks like duplicated code.

File: lib/domain_viewer_tracker/viewer.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+#  id         :integer          not null, primary key
+#  uuid       :string(255)
+#  user_id    :integer
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+# Indexes
+#
+#  index_viewers_on_uuid_and_user_id  (uuid,user_id) UNIQUE
+#
```
Comment (chiastolite) [inline lib/domain_viewer_tracker/viewer.rb]: Is this comment necessary?

File: spec/domain_viewer_tracker_spec.rb (hunk 1 above)
Comment (ryonext) [inline spec/domain_viewer_tracker_spec.rb]: Ok, I'll fix it :smile:

File: spec/domain_viewer_tracker_spec.rb (hunk 2 above)
Comment (ryonext) [inline spec/domain_viewer_tracker_spec.rb]: Ok, I'll fix it :smile:

File: lib/domain_viewer_tracker/viewer.rb (hunk 3 above)
Comment (ryonext) [inline lib/domain_viewer_tracker/viewer.rb]: Ok, I'll fix it :smile:

**Comment by nozayasu** [comment]
:+1:

**Comment by chiastolite** [comment]
:eyes:

**Pull request merged**